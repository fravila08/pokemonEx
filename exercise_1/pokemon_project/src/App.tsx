import { useEffect, useState } from 'react'
import reactLogo from './assets/react.svg'
import './App.css'
import { Container } from 'react-bootstrap'
import axios from 'axios'


export const getPokemon = async() =>{
  let response = await axios.get("https://pokeapi.co/api/v2/pokemon/ditto")
  return { 
    name : response.data['name'],
    img : response.data['sprites']['front_default']
  }
}

function App() {
  const [pokemon, setPokemon] = useState({})
  
  useEffect(()=>{
    const myPokemon =async () => {
      let pokeObject = await getPokemon()
      setPokemon(pokeObject)
    }
    myPokemon()
  },[])

  useEffect(()=>{
    console.log(pokemon)
  }, [pokemon])

  return (
    <Container className="App">
      <h1 id='titleHeader'>Pokemon API EX</h1>
    </Container>
  )
}

export default App
