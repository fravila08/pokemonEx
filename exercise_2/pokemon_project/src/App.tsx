import { useEffect, useState } from 'react'
import './App.css'
import { Container } from 'react-bootstrap'
import axios from 'axios'
import { PokemonCard } from './components/pokemonCard'

export interface IPokemon {
  name : string;
  img : string; 
}


export async function getPokemon(): Promise<IPokemon>{
  let response = await axios.get("https://pokeapi.co/api/v2/pokemon/ditto")
  return { 
    name : response.data['name'],
    img : response.data['sprites']['front_default']
  }
}

function App() {
  const [pokemon, setPokemon] = useState<IPokemon>({name:'none', img:'none'})
  const [show, setShow] = useState(false)
  useEffect(()=>{
    const myPokemon =async () => {
      let pokeObject = await getPokemon()
      setPokemon(pokeObject)
      setShow(true)
    }
    myPokemon()
  },[])

  useEffect(()=>{
    console.log(pokemon)
  }, [pokemon])

  return (
    <Container className="App">
      <h1 id='titleHeader'>Pokemon API EX</h1>
      {show ?
       <PokemonCard pokemon={pokemon} />
       :
       null
      }
    </Container>
  )
}

export default App
