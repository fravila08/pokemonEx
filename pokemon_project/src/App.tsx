import { useState } from 'react'
import reactLogo from './assets/react.svg'
import './App.css'
import { Container } from 'react-bootstrap'

function App() {
  

  return (
    <Container className="App">
      <h1>Pokemon API EXAMPLE</h1>
    </Container>
  )
}

export default App
