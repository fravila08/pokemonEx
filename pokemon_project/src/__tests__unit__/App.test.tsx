import {describe, it , expect} from "vitest";
import TestRenderer from 'react-test-renderer';
import App from "../App";

describe("App", () => {

    it("create html snapshot", () => {
        const app = TestRenderer.create(<App />)
        expect(app).toMatchSnapshot();
    })

})