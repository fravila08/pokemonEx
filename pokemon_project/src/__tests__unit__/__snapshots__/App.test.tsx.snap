// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`App > create html snapshot 1`] = `
<div
  className="App container"
>
  <h1>
    Pokemon API EXAMPLE
  </h1>
</div>
`;
